pygsti/forwardsims/torchfwdsim.py

```diff
@@ -19,206 +25,240 @@
     from pygsti.layouts.copalayout import CircuitOutcomeProbabilityArrayLayout
     import torch
 
-import numpy as np
 import warnings as warnings
-from collections import OrderedDict
 from pygsti.modelmembers.torchable import Torchable
 from pygsti.forwardsims.forwardsim import ForwardSimulator
 
+try:
+    import torch
+    TORCH_ENABLED = True
+except ImportError:
+    TORCH_ENABLED = False
+    pass
+
 
 class StatelessCircuit:
     """
-    Helper data structure useful for simulating a specific circuit quantum (including prep,
+    Helper data structure useful for simulating a specific quantum circuit (including prep,
     applying a sequence of gates, and applying a POVM to the output of the last gate).
     
     The forward simulation can only be done when we have access to a dict that maps
-    pyGSTi Labels to certain PyTorch Tensors.
+    pyGSTi Labels to certain PyTorch Tensors. Construction of that dict is managed by the
+    StatelessModel class, defined below.
     """
 
     def __init__(self, spc: SeparatePOVMCircuit):
         self.prep_label = spc.circuit_without_povm[0]
-        self.op_labels  = spc.circuit_without_povm[1:]
+        if len(spc.circuit_without_povm) > 1:
+            self.op_labels  = spc.circuit_without_povm[1:]
+        else:
+            # Importing this at the top of the file would create a circular dependency.
+            from pygsti.circuits.circuit import Circuit
+            self.op_labels = Circuit(tuple())
         self.povm_label = spc.povm_label
+        self.outcome_probs_dim = len(spc.effect_labels)
```

Comment:
Note for future development. I think this should be fine for circuits with only POVMS (as is currently supported), but I don't believe this will give you the correct value when instruments are present. OpModels have a method called `circuit_outcomes` which should handle the legwork of figuring this all out when instruments are present, though (it is more complicated to implement this than it seems, so it is always good when someone else figured it out already). Not action items now, just something to keep in mind whenever we decide to implement torchable instruments.